extensions/wikia/AdEngine/js/spec/video/videoFrequencyMonitor.spec.js

```diff
@@ -0,0 +1,275 @@
+/*global describe, it, expect, modules, spyOn*/
+describe('ext.wikia.adEngine.video.videoFrequencyMonitor', function () {
+	'use strict';
+
+	var noop = function () {},
+		mocks = {
+			store: {
+				save: noop,
+				numberOfVideosSeenInLastPageViews: noop,
+				numberOfVideosSeenInLast: noop
+			},
+			adLogicPageViewCounter: {
+				get: function () {
+					return 1;
+				}
+			},
+			settings: {
+				get: function () {
+					return {
+						pv: [],
+						time: []
+					};
+				}
+			}
+		};
+
+	beforeEach(function () {
+		jasmine.clock().install();
+	});
+
+	afterEach(function () {
+		jasmine.clock().uninstall();
+	});
+
+	function mockVideosOnPageViews(data) {
+		if (!mocks.store.numberOfVideosSeenInLastPageViews.and) {
```

Comment:
This looks like a creative structure for mocking. Can we simplify it and call `spyOn` only once?